packages/ckeditor5-table/src/commands/inserttablecommand.js

```diff
@@ -49,9 +49,21 @@ export default class InsertTableCommand extends Command {
 		const model = this.editor.model;
 		const selection = model.document.selection;
 		const tableUtils = this.editor.plugins.get( 'TableUtils' );
+		const config = this.editor.config.get( 'table' );
 
 		const insertionRange = findOptimalInsertionRange( selection, model );
 
+		const defaultRows = config.defaultHeadings.rows;
+		const defaultColumns = config.defaultHeadings.columns;
+
+		if ( options.headingRows === undefined && defaultRows ) {
+			options.headingRows = Math.min( defaultRows, options.rows );
```

Comment:
In here you don't know what are the default values for rows and columns, those could be undefined here. Please move clamping to the `TableUtils#createTable()` because it's where those default rows/columns are resolved.